src/_pytest/tmpdir.py

```diff
@@ -172,7 +172,9 @@ def _mk_tmp(request: FixtureRequest, factory: TempPathFactory) -> Path:
 def tmpdir(tmp_path: Path) -> py.path.local:
     """Return a temporary directory path object which is unique to each test
     function invocation, created as a sub directory of the base temporary
-    directory.
+    directory. After 3 sessions the directory will be removed by default,
```

Comment:
I think it would be good to describe the situation in a bit more detail. I suggest writing this in a separate paragraph:

```
The temporary directory is created under a base directory. By default, a
new base is created each test session, and old bases are removed after 3
sessions, to aid in debugging. If ``--basetemp`` is used then it is
cleared each session. See :ref:`base temporary directory`.
```

WDYT?

Reply:
Thanks, that sounds good, particularly the extra link! I updated with 6a25660. I left out your first sentence since I thought it was sufficiently covered in the existing docs, but I can be persuaded otherwise.